polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -911,18 +920,52 @@ function IndexTableBase({
         ? sortDirection!
         : defaultSortDirection;
 
-      const tooltipContent = sortToggleLabels[index][tooltipDirection];
+      const sortTooltipContent = sortToggleLabels[index][tooltipDirection];
+
+      if (!heading.tooltipContent) {
+        return (
+          <Tooltip {...defaultTooltipProps} content={sortTooltipContent}>
+            {sortMarkup}
+          </Tooltip>
+        );
+      }
 
       return (
-        <Tooltip
-          content={tooltipContent}
-          activatorWrapper="div"
-          dismissOnMouseOut
-        >
-          {sortMarkup}
-        </Tooltip>
+        <div className={styles.SortableTableHeadingWithCustomMarkup}>
+          <UnstyledButton
+            onClick={() => handleSortHeadingClick(index, newDirection)}
+            className={styles.TableHeadingSortButton}
+            tabIndex={selectMode ? -1 : 0}
+          >
+            {headingContent}
+
+            <Tooltip
+              {...defaultTooltipProps}
+              content={sortTooltipContent}
+              preferredPosition="above"
+            >
+              {iconMarkup}
+            </Tooltip>
+          </UnstyledButton>
+        </div>
       );
     }
+
+    if (heading.tooltipContent) {
+      <Tooltip
+        {...defaultTooltipProps}
+        content={heading.tooltipContent}
+        padding="4"
+        borderRadius="2"
+        preferredPosition="above"
+        hasUnderline
+      >
+        <span className={styles.TableHeadingSortIconWrapper}>
+          {headingContent}
+        </span>
+      </Tooltip>;
+    }
```

Comment:
This is a non-sortable header that has the `heading.tooltipContent` prop. This enables this header to have an underline by default, as well as configure the style of the header.